# Python_Pattern_Counting

This script can be used as a tool to recognize and count different patterns in a DNA sequence. This script was made for an assignment in our Big Data Class:

Write a python script to:

1. Count the number of each base across all sequences.
2. Count the number of each pair of bases (described as kmers of size 2).
3. Count the number of each pair of bases, without  assuming you know in advance the possible pairs (i.e. start with an empty dictionary).
4. Alter the counting function to allow for counting kmers of any size.
5. Organize your Jupyter notebook so that you have imported packages at the top, then your functions (count_2kmer; count2kmer_unknown; count_kmers), then a block where each function is called and outputs results.
6. Make sure your code is commented and functions have docstrings.

A sample dataset is at http://rachelss.github.io/Bioinformatics/sample.fastq
